File: apps/android/app/src/main/java/com/alafteknoloji/photon/core/designsystem/components/PhotonLogoMark.kt
package com.alafteknoloji.photon.core.designsystem.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import kotlin.math.PI
import kotlin.math.cos
import kotlin.math.sin

/**
 * Elegant, transparent vector brandmark for Photon.
 * 100% Parity with iOS PhotonLogoMark.swift.
 */
@Composable
fun PhotonLogoMark(
    size: Dp = 32.dp,
    color: Color = Color.White,
    modifier: Modifier = Modifier
) {
    Canvas(modifier = modifier.size(size)) {
        val center = Offset(size.toPx() / 2f, size.toPx() / 2f)
        val radius = minOf(size.toPx(), size.toPx()) / 2f
        val strokeWidth = radius * 0.12f

        // Outer optical ring
        drawCircle(
            color = color,
            radius = radius - strokeWidth / 2f,
            center = center,
            style = Stroke(width = strokeWidth)
        )

        // 6-blade minimalist aperture rays
        val bladeCount = 6
        val innerRadius = radius * 0.36f
        val rayLength = radius * 0.78f

        for (i in 0 until bladeCount) {
            val angle = (i * (2.0 * PI / bladeCount) - (PI / 6.0)).toFloat()
            val tangentAngle = (angle + (PI / 3.2)).toFloat()

            val startX = center.x + cos(angle) * innerRadius
            val startY = center.y + sin(angle) * innerRadius
            val endX = startX + cos(tangentAngle) * (rayLength - innerRadius)
            val endY = startY + sin(tangentAngle) * (rayLength - innerRadius)

            drawLine(
                color = color,
                start = Offset(startX, startY),
                end = Offset(endX, endY),
                strokeWidth = strokeWidth * 0.9f,
                cap = StrokeCap.Round
            )
        }

        // Central photon light core point
        val coreRadius = radius * 0.14f
        drawCircle(
            color = color,
            radius = coreRadius,
            center = center
        )
    }
}

File: apps/ios/Photon/Core/DesignSystem/Components/PhotonLogoMark.swift
//
//  PhotonLogoMark.swift
//  Photon
//
//  Created by Metin TASCI on 14.08.2026.
//

import SwiftUI

/// Elegant, transparent vector brandmark for Photon.
/// Features a minimalist optical aperture & photon ray geometry with zero opaque background.
public struct PhotonLogoMark: View {
    let size: CGFloat
    let color: Color
    
    public init(size: CGFloat = 32, color: Color = .primary) {
        self.size = size
        self.color = color
    }
    
    public var body: some View {
        Canvas { context, canvasSize in
            let center = CGPoint(x: canvasSize.width / 2, y: canvasSize.height / 2)
            let radius = min(canvasSize.width, canvasSize.height) / 2
            let strokeWidth = radius * 0.12
            
            // Outer optical ring
            let outerRect = CGRect(
                x: center.x - radius + strokeWidth / 2,
                y: center.y - radius + strokeWidth / 2,
                width: (radius - strokeWidth / 2) * 2,
                height: (radius - strokeWidth / 2) * 2
            )
            context.stroke(
                Path(ellipseIn: outerRect),
                with: .color(color),
                lineWidth: strokeWidth
            )
            
            // 6-blade minimalist aperture rays
            let bladeCount = 6
            let innerRadius = radius * 0.36
            let rayLength = radius * 0.78
            
            for i in 0..<bladeCount {
                let angle = Double(i) * (2 * Double.pi / Double(bladeCount)) - (Double.pi / 6)
                let tangentAngle = angle + (Double.pi / 3.2)
                
                let startX = center.x + CGFloat(cos(angle)) * innerRadius
                let startY = center.y + CGFloat(sin(angle)) * innerRadius
                let endX = startX + CGFloat(cos(tangentAngle)) * rayLength
                let endY = startY + CGFloat(sin(tangentAngle)) * rayLength
                
                var path = Path()
                path.move(to: CGPoint(x: startX, y: startY))
                path.addLine(to: CGPoint(x: endX, y: endY))
                
                context.stroke(
                    path,
                    with: .color(color),
                    style: StrokeStyle(lineWidth: strokeWidth * 0.9, lineCap: .round)
                )
            }
            
            // Central photon light core point
            let coreRadius = radius * 0.14
            let coreRect = CGRect(
                x: center.x - coreRadius,
                y: center.y - coreRadius,
                width: coreRadius * 2,
                height: coreRadius * 2
            )
            context.fill(Path(ellipseIn: coreRect), with: .color(color))
        }
        .frame(width: size, height: size)
    }
}

#Preview {
    ZStack {
        Color.black.ignoresSafeArea()
        VStack(spacing: 20) {
            PhotonLogoMark(size: 80, color: .white)
            PhotonLogoMark(size: 40, color: .cyan)
        }
    }
}
